futures-test/src/task/wake/noop.rs

```diff
@@ -18,14 +18,12 @@ use std::sync::Arc;
 /// cx.waker().wake();
 /// ```
 #[derive(Debug)]
-pub struct Noop {
-    _reserved: (),
-}
+pub struct Noop {}
 
 impl Noop {
     /// Create a new instance
     pub fn new() -> Self {
-        Self { _reserved: () }
+        Self {}
```

Comment:
This might work as just `Self`, otherwise just `Noop`

Reply:
You could also just drop the `new` function since anyone can construct a field-less ZST.